servicetalk-http-api/src/test/java/io/servicetalk/http/api/ContentEncodingHttpServiceFilterTest.java

```diff
@@ -73,4 +87,46 @@ void testMatchAndRemoveEncodingMiddleDoesNotMatch() {
         assertThat("unexpected header: " + contentEncoding,
                 contentEqualsIgnoreCase(contentEncoding, " deflate , foo , gzip "), is(true));
     }
+
+    @ParameterizedTest(name = "{displayName} [{index}] {0}-{1}")
+    @MethodSource("requesterTypes")
+    void contentLengthRemovedOnEncode(final RequesterType type,
+                                      final SecurityType security) throws Exception {
+        setUp(security);
+        StreamingHttpRequester filter = createFilter(type, (respFactory, request) -> {
+            try {
+                assertContentLength(request.headers(), request.payloadBody());
+            } catch (Exception e) {
+                return failed(e);
+            }
+
+            // Simulate the server compressed reply, with CONTENT_LENGTH included.
+            String responseStr = "aaaaaaaaaaaaaaaa";
+            Buffer responseBuf = DEFAULT_ALLOCATOR.fromAscii(responseStr);
+            Buffer encoded = gzipDefault().encoder().serialize(responseBuf, DEFAULT_ALLOCATOR);
+            return Single.succeeded(respFactory.ok().payloadBody(from(encoded))
+                    .addHeader(CONTENT_LENGTH, String.valueOf(encoded.readableBytes()))
+                    .addHeader(CONTENT_ENCODING, gzipDefault().encodingName()));
+        }, new ContentEncodingHttpRequesterFilter(new BufferDecoderGroupBuilder()
+                .add(gzipDefault(), true)
+                .add(identityEncoder(), false)
+                .build()));
+
+        // Simulate the user (or earlier filter) setting the content length before compression.
+        StreamingHttpRequest request = filter.post("/foo");
+        String payloadBody = "bbbbbbbbbbbbbbbbbbb";
+        request.payloadBody(from(filter.executionContext().bufferAllocator().fromAscii(payloadBody)));
+        request.headers().add(CONTENT_LENGTH, String.valueOf(payloadBody.length()));
+        request.contentEncoding(gzipDefault());
+        StreamingHttpResponse response = filter.request(request).toFuture().get();
+        assertContentLength(response.headers(), response.payloadBody());
+    }
+
+    private static void assertContentLength(HttpHeaders headers, Publisher<Buffer> publisher) throws Exception {
+        CharSequence reqLen = headers.get(CONTENT_LENGTH);
+        if (reqLen != null) {
+            final int len = publisher.collect(() -> 0, (sum, buff) -> sum + buff.readableBytes()).toFuture().get();
```

Comment:
The test never goes inside this `if` branch. Instead, consider asserting that there is not CL header bcz this is what we expect from a streaming client.

Reply:
I'll just leave as is for now ... currently we just remove the header but from the test perspective either it is not present, or it needs to be correct.